### ISSUE TITLE
[L4D2] Medal achievements are bugged

### ISSUE THREAD

**Issue by Dragons952** [ValveSoftware/Source-1-Games#3441]
On some survival maps after completing the tasks required to unlock the achievement; it doesn't give me the check mark for completion. 
![Bugged killing spree achievement](https://user-images.githubusercontent.com/69691675/99142028-59a03000-260e-11eb-85fa-530bac3713b9.png)

Even after unlocking the achievements, the check marks for completion disappear on some of them.
![Bronze](https://user-images.githubusercontent.com/69691675/99142048-966c2700-260e-11eb-976c-55a51eb35d23.PNG)
![Silver](https://user-images.githubusercontent.com/69691675/99142050-9a984480-260e-11eb-8be6-1c070612532e.PNG)

System info:
Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz

Operating System Version:
    Windows 10 (64 bit)
    NTFS:  Supported
    Crypto Provider Codes:  Supported 311 0x0 0x0 0x0

Video Card:
    Driver:  Intel(R) HD Graphics 530
    DirectX Driver Name:  nvldumd.dll
    Driver Version:  27.20.100.8935
    DirectX Driver Version:  27.21.14.5655
    Driver Date: 10 28 2020
    OpenGL Version: 4.6
    Desktop Color Depth: 32 bits per pixel
    Monitor Refresh Rate: 60 Hz
    DirectX Card: NVIDIA GeForce GTX 960M
    VendorID:  0x10de
    DeviceID:  0x139b
    Revision:  0xa2
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    No SLI or Crossfire Detected
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 15.00" x 8.43" (17.17" diag)
                                            38.1cm x 21.4cm (43.6cm diag)
    Primary Bus: PCI Express 8x
    Primary VRAM: 1024 MB
    Supported MSAA Modes:  2x 4x 8x 

Memory:
    RAM:  16282 Mb

**Comment by sebeg778** [comment]
I have the same problem in **Left 4 Dead (1)**. I managed to survive required 4 minutes in all 16 maps for bronze medal. The achievement menu section shows I completed 15/16 maps. There's no checkmark ✔ on the **Bridge (Crash Course)** map, even though I survived over 4 minutes for multiple times. The game even shows my record time which is something about 6 minutes.

**Comment by DeWillem78** [comment]
Hi! I have similar issues with these achievements (l4d2). I have gold on all official survival maps. Confirmed when loading each of the maps/chapters. In-game at the achievements menu it shows these bronze, silver, gold achievements as 34/43 with several check marks missing in each one. Last few weeks I've been obtaining gold for all those maps. Also in-game at the leaderboards menu it shows my gold times in all the lists except for the three lists from the blood harvest campaign where I get the notification: "No records were found for this leaderboard."

On the steamcommunity.com website however, it shows 40/43 bronze, silver and gold achievements. So the website and in-game achievements overviews aren't synchronized. And neither one is complete and accurate. I would be very grateful to see these achievements fixed and obtainable.

Will of course supply PC specs or other relevant information if necessary.

Cheers,
Willem

**Comment by DeWillem78** [comment]
After installing a small update today, I loaded and searched some survival maps looking for golden crowbars. And much to my surprise I found the number of completed gold chapters increasing. I was able to load the chapters that were incorrectly missing a checkmark en complete all three related achievements. I can't find a change list but if this was fixed in the past two days, thank you!

Cheers,
Willem
